File: Examples/CallGraph/TypeScript/OrderController.ts
import { Order } from "./Order";
import { OrderService } from "./OrderService";
import { Validator } from "./Validator";

/** Entry point: fans out to validation and order placement. */
export class OrderController {
    constructor(
        private readonly service: OrderService,
        private readonly validator: Validator
    ) {}

    submit(order: Order): void {
        this.validator.validate(order);
        this.service.place(order);
    }
}

File: Examples/CallGraphDiff/Java/Before/OrderController.java
/** Entry point: fans out to validation and order placement. */
class OrderController {
    private final OrderService service;
    private final Validator validator;

    OrderController(OrderService service, Validator validator) {
        this.service = service;
        this.validator = validator;
    }

    void submit(Order order) {
        validator.validate(order);
        service.place(order);
    }
}
